frontend/src/Components/SearchBar/SearchBar.jsx

```diff
@@ -1,90 +1,82 @@
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useEffect } from "react";
+import debounce from "lodash.debounce";
 import "./SearchBar.css";
 
-const SearchBar = ({ onSelect, onSearchOnMap }) => {
+function SearchBar({ userId, onSelect }) {
   const [query, setQuery] = useState("");
-  const [results, setResults] = useState([]);
-  const [showDropdown, setShowDropdown] = useState(false);
-  const latestRequest = useRef(null);
+  const [suggestions, setSuggestions] = useState([]);
 
-  const handleInputChange = async (e) => {
-    const value = e.target.value;
-    setQuery(value);
+  const fetchSuggestions = debounce(async (input) => {
+    if (input.length > 0) {
+      try {
+        const response = await fetch(
+          `http://localhost:3000/api/search?query=${input}&userId=${userId}`
+        );
 
-    if (value.trim() === "") {
-      setResults([]);
-      setShowDropdown(false);
-      return;
-    }
-
-    if (latestRequest.current) {
-      latestRequest.current.abort();
-    }
+        if (!response.ok) {
+          throw new Error("Network response was not ok");
+        }
 
-    const controller = new AbortController();
-    latestRequest.current = controller;
+        const data = await response.json();
 
-    try {
-      const response = await fetch(
-        `http://localhost:3000/api/businesses/search?query=${value}`,
-        { signal: controller.signal }
-      );
-      if (!response.ok) {
-        throw new Error("Failed to fetch search results");
-      }
-      const data = await response.json();
-      setResults(data);
-      setShowDropdown(true);
-    } catch (err) {
-      if (err.name !== "AbortError") {
-        console.error(err);
-        setResults([]);
-        setShowDropdown(false);
+        if (Array.isArray(data)) {
+          setSuggestions(data);
+        } else {
+          setSuggestions([]);
+        }
+      } catch (error) {
+        setSuggestions([]);
       }
+    } else {
+      setSuggestions([]);
     }
+  }, 300);
```

Comment:
nit: i'd bump this down to around 150 or 200 ms to avoid slowing down the search time